### ISSUE TITLE
Allow customization of the supabase root directory name

### ISSUE THREAD

**Issue by Bowbee** [supabase/cli#2139]
**Is your feature request related to a problem? Please describe.**
I have a monorepo that has a very specific naming structure with `/apps` and `/libs` folders. The option for `--workdir` is great when initializing a new project, however I would love to specify the name of the root directory alongside this.

**Describe the solution you'd like**
Having a cli flag for `--rootname <folder name>` would be great.

**Describe alternatives you've considered**
The alternative is to place a parent folder with the desired name before initializing a project. Eg: `apps/testproject/supabase`.
Something like this is required for multiple projects to exist within a monorepo.

**Additional context**
This appears to currently be hardcoded in [internal/utils/misc.go](https://github.com/supabase/cli/blob/develop/internal/utils/misc.go#L169).
I am unsure whether other considerations are needed such as an indicator for what is a supabase directory or not. Maybe the CLI can automatically detect this based off `--workdir` or the contents of the `cwd`.

**Comment by sweatybridge** [comment]
Do you mean to customize like `apps/testproject/custom-supabase`? That wouldn't be possible because a lot of our platform's GitHub integrations depend on a directory named `supabase`.

If you mean `apps/testproject/supabase`, I think it's already supported with `supabase init --workdir apps/testproject` flag.

**Comment by Bowbee** [comment]
Yes I mean `apps/testpropject/custom-supabase` - this would be great. Maybe this can be accomplished by having the directory name specified in the `config.toml` instead of assumed to be `supabase` always?

**Comment by sweatybridge** [comment]
`config.toml` lives inside `supabase` directory though. How would you suggest we find the right config.toml file if we support `custom-supabase` directory?

**Comment by tomx-sh** [comment]
@sweatybridge I'm also interested in the possibility to place Supabase directories anywhere I want in my project.
Next.JS docs encourage a structure where everything is in the `src` or `app` folder, and the root directory is only used for config files.
I can imagine a `supabase.config.toml` file here, or a `.supabase` directory ' (personally prefer the first solution) where you would provide the paths of supabase stuff!

**Comment by sweatybridge** [comment]
I see, so perhaps a `--config <path-to-toml>` flag that can be passed to CLI to override the default location of `supabase/config.toml`.

Within `config.toml`, users can declare path overrides for other directories relative to the location of `config.toml`. For eg.

```toml
workdir = '..'

[migrations]
path = './migrations/*.sql'

[seed]
path = './seed.sql'

[functions.slug]
path = '../src/index.ts'
```

**Comment by baffioso** [comment]
> I see, so perhaps a `--config <path-to-toml>` flag that can be passed to CLI to override the default location of `supabase/config.toml`.
> 
> Within `config.toml`, users can declare path overrides for other directories relative to the location of `config.toml`. For eg.
> 
> ```toml
> workdir = '..'
> 
> [migrations]
> path = './migrations/*.sql'
> 
> [seed]
> path = './seed.sql'
> 
> [functions.slug]
> path = '../src/index.ts'
> ```

This could indeed be handy. I using a monorepo for multiple identical apps with the same supabase setup and would like to have separate config for each project, but use the same migrations/seed etc. This would probably do the job.

**Comment by djhi** [comment]
> I see, so perhaps a --config <path-to-toml> flag that can be passed to CLI to override the default location of supabase/config.toml.

This would also be handy to have separate containers and volumes for running e2e tests without touching your development instances.

**Comment by Donnerstagnacht** [comment]
I would like to have a feature in which I can simply reset my local database once with my declarative schemas, e.g. instead of diffing it, just apply all declared schemas (sql files) as migrations once.

I think that would be a handy option keep sql code of early projects organized.

Something like db reset --use-schemas or db apply or db apply schemas.

**Comment by IdanDavidAviv** [comment]
what about changing this dir's name to `.supabase` as the dir name? 
git cursor and pycharm use it

**Comment by madsem** [comment]
This is unexpected, I have a monorepo as well and the need for multiple supabase dbs in the same project, I was wondering why I'm seeing all these errors in my failing CI flow, after renaming `supabase/` to `customer-db/`.

Really expected this to be working, how else can I have multiple db's in one project?

**Comment by IdanDavidAviv** [comment]
@madsem what i ended up doing is creating a git project for each db. might be a little hassle to migrate, but it was worth it in my case.

**Comment by madsem** [comment]
> [@madsem](https://github.com/madsem) what i ended up doing is creating a git project for each db. might be a little hassle to migrate, but it was worth it in my case.

That could def be an option, sadly not for our use case. Currently thinking of just going the route of least resistance, and just nest one more level:

```
customer-db/
├── README.md
└── supabase/
    ├── config.toml
    ├── migrations/
    └── seed.sql

analytics/
├── README.md
└── supabase/
    ├── config.toml
    └── migrations/
```

etc.

**Comment by IdanDavidAviv** [comment]
@madsem 100%, would've done the same if separation was not sensible for us
if you use supabase CLI it should also recognize that you have 2 projects, you can set both locally \w docker, just make sure you use different ports so both work back to back locally
good luck!

**Comment by imtreydp** [comment]
@sweatybridge can we please get this added? you seem to have already [agreed to adding this functionality](https://github.com/supabase/cli/pull/3092#pullrequestreview-2591945816) and there's [an open PR](https://github.com/supabase/cli/pull/3092) that implements it. 

could you take a look and give an update? thanks in advance!

**Comment by leen-neel** [comment]
@sweatybridge I'd like to take this if the design is still the one you sketched here.

First slice, matching your comments:
- add `--config <path>` so the CLI isn't hardcoded to `supabase/config.toml`
- derive workdir from that config path (the gap you called out on #3092)
- allow path overrides in the toml relative to the config file (migrations, seed, functions)
- keep today's `supabase/` layout as the default

Could this get `open-for-contribution` so a PR won't be auto-closed? Happy to wait if the scope should be smaller.
